soccer/src/soccer/strategy/agent/position/penalty_player.cpp

```diff
@@ -43,14 +56,41 @@ PenaltyPlayer::State PenaltyPlayer::update_state() {
 
 std::optional<RobotIntent> PenaltyPlayer::state_to_task(RobotIntent intent) {
     switch (latest_state_) {
-        case LINE_UP: {
+        case START: {  // First, gets the robot to the ball to begin penalty dribbling-shooting
+            SPDLOG_INFO("START");
+            double y_pos = last_world_state_->ball.position.y();
+            y_pos -= kRobotRadius + 0.3;  // added the 0.01 as a buffer space
+            rj_geometry::Point target_pt{last_world_state_->ball.position.x(), y_pos};
+            rj_geometry::Point target_vel{0.0, 0.0};
+            // Face ball
+            planning::PathTargetFaceOption face_option{planning::FaceBall{}};
+
+            // Create Motion Command
+            planning::LinearMotionInstant goal{target_pt, target_vel};
+            intent.motion_command =
+                planning::MotionCommand{"path_target", goal, planning::FaceBall{}};
+            break;
+        }
+        case SMALL_KICK: {  // less of a kick, more of a "follow" ball closely
+            // SPDLOG_INFO("SMALL_KICK");
+            rj_geometry::Point center_goal = field_dimensions_.their_goal_loc();
+            auto line_kick_cmd =
+                planning::MotionCommand{"line_kick", planning::LinearMotionInstant{center_goal}};
+
+            intent.motion_command = line_kick_cmd;
+            intent.shoot_mode = RobotIntent::ShootMode::KICK;
+            intent.trigger_mode = RobotIntent::TriggerMode::ON_BREAK_BEAM;
+            intent.kick_speed = 0.0;
+            // the point of making a 0 kick speed is to fake dribble since we cannot get the vaccum
+            // behavior to work
+
+            return intent;
+        }
+        case LINE_UP: {  // gets the robot behind the ball with a certain distance (usually
+                         // immediatly skipped if the robot is already close)
+            // SPDLOG_INFO("LINE_UP");
             double y_pos = last_world_state_->ball.position.y();
-            // if ball is above goal, increase y_pos, else decrease
-            // if (y_pos - field_dimensions_.their_goal_loc().y() > 0) {
-            // y_pos += kRobotRadius - 0.15;
-            // } else {
-            y_pos -= kRobotRadius + 0.3;
-            // }
+            y_pos -= kRobotRadius + 0.1;
```

Comment:
What is the 0.1 doing? Can we relate this to the radius as a ratio or something?

Reply:
do you mean ```kOwnBallRadius``` possibly?